api/agent/listeners.go

```diff
@@ -0,0 +1,37 @@
+package agent
+
+import (
+	"context"
+
+	"github.com/fnproject/fn/api/extensions"
+	"github.com/fnproject/fn/api/models"
+)
+
+type callTrigger interface {
```

Comment:
would be kinda sexy to have this just be a `CallListener` that has a `[]CallListener` and it calls each in `Before` and `After`, but an unimportant nit; it would remove an extra type of `callTrigger` that is getting carried around, at least.

Reply:
There's already the CallListener interface.

Reply:
right, I guess to be clear, it seems that `callTrigger` could pretty easily implement `CallListener` to do its work through those methods rather than having the additional `fireBeforeDispatch` and `fireAfterDispatch` methods (they are similar in nature).